akka-actor-testkit-typed/src/main/scala/akka/actor/testkit/typed/scaladsl/ActorTestKit.scala

```diff
@@ -185,6 +186,24 @@ final class ActorTestKit private[akka] (val name: String, val config: Config, se
    */
   def createTestProbe[M](name: String): TestProbe[M] = TestProbe(name)(system)
 
+  /**
+   * Verify serialization roundtrip.
+   * Throws exception from serializer if `obj` can't be serialized and deserialized.
+   *
+   * @param obj the object to verify
+   * @param assertEquality if `true` the deserialized  object is verified to be equal to `obj`,
+   *                       and if not an `AssertionError` is thrown
+   * @return the deserialized object
+   */
+  def verifySerialization[M](obj: M, assertEquality: Boolean): M = {
+    import akka.actor.typed.scaladsl.adapter._
+    val result =
+      SerializationExtension(internalSystem.toUntyped).verifySerialization(obj.asInstanceOf[AnyRef]).asInstanceOf[M]
+    if (assertEquality && result != obj)
+      throw new AssertionError(s"verifySerialization expected $obj, but was $result")
+    result
+  }
```

Comment:
I mostly added it because it's very useful when testing EventSourcedBehaviors, e.g. serialization of state (snapshot) isn't easy to simulate. Ok, I can place it in a new `SerializationTestKit` but would still be nice to access it via `ActorTestKit` to not have several things to bring in.

Reply:
moved in e513a71

Reply:
👍